README.md

```diff
@@ -16,44 +16,79 @@ The rough workflow for this is, per barcode:
 2.  Hit [our platform api](https://platformdocs.nypl.org/#/recap/get_v0_1_recap_nypl_bibs) with the customer code and barcode and receive back SCSBXML.
 3.  Do minor massaging and post updated information to the [SCSB "submit collection" endpoint](https://uat-recap.htcinc.com:9093/swagger-ui.html#!/shared-collection-rest-controller/submitCollection).
 
-After the updates - this app will, in some way, shape or form send an email to
-the staff who initiated the update. (Success / failure)
+## Application Architecture
+
+### High Level Workflow:
+
+This application...
+
+1.  Consumes messages from Amazon SQS.
+2.  Persists those messages into Redis, and delete them from SQS.
+3.  Works messages from Redis using [resque](https://github.com/resque/resque) in a separate process.
+
+### Details
+
+#### From SQS -> Redis
+
+1.  `dequeue_from_sqs.rb` consumes messages from SQS. It does that by instantiating an `SQSMessageHandler`.
+1.  `SQSMessageHandler` makes sure that a message is well formed & old enough to be worked.  
+If it is, it's persisted into Redis and deleted from SQS.
+
+#### Working Redis
+
+1.  `ProcessResqueMessage`, a [resque job](https://github.com/resque/resque#overview), uses an instance of `ResqueMessageHandler` do all the hard work.
+1.  `ResqueMessageHandler` is what actually inspects the message, makes the appropriate API calls (through other classes), and conditionally sends error reports.
 
 ## Installing & Running
 
 ### Locally
 
 #### Setup
 
+1.  Ensure you have Redis installed & running on your machine (`brew install redis`)
 1.  `cp ./config/.env.example ./config/.env`
 1.  `gem install bundler --version 1.16.1`
 1.  `bundle install`
 
-#### Usage
+#### Running Natively Locally
 
-`ruby ./consume_messages.rb`
+1. `ruby ./dequeue_from_sqs.rb` and in another tab...`QUEUE=* rake resque:work`
+1.  Make sure the environment variable of `IS_DRY_RUN` is set correctly. If set to false, it will update the incomplete barcodes with SCSBXML in the assigned ReCap environment. If set to true, it will run the script without updating the barcodes.
 
-### Docker
+#### Running From Docker Locally
 
-#### Building a Docker Image
+You can use docker and [`docker-compose`](https://docs.docker.com/compose/overview/) to run the app locally too.
```

Comment:
That's a good point - there's certainly room for refactoring the README.
I'm just not sure if I'm going to end up running natively or through docker-compose more often.
docker-compose requires building a new docker image per change which _coulllldddddd_ be an annoyingly long feedback loop.

Reply:
Hmmmm... it shouldn't need an image rebuild for a code change. There is a way to configure the image so that it is reading directly (not copying) from the local hard-drive. Which takes a lot of the pain out of using docker. Check out point 5 in https://docs.docker.com/compose/gettingstarted/#step-5-edit-the-compose-file-to-add-a-bind-mount.

I feel like we should at some point get to the point where we say — "develop in Docker" or "develop locally but run tests in Docker" or whatever. But it's fine to leave it indeterminate while we're finding our way.

Reply:
I think we're a _hair_ away from being able to use mount points to develop & test locally.
There would be a few things that need to change in the `Dockerfile`.

Off the top of my head we'd probably need to use [`--build-arg`](https://docs.docker.com/engine/reference/commandline/build#set-build-time-variables-build-arg) to dictate what args we pass to `bundle install` so we don't install dev/test dependencies in production or forego them locally.

But again - I think we're really close to always being able to test (including integration tests using the docker-compose's Redis instance!) and develop locally.

This week is so crazy time-wise that I may just merge it in and follow up in my free time with a change to the `Dockerfile`, `docker-compose.yml`, & README about a way to stay in container-world.

Reply:
Sounds fantastic — thanks!

I'd definitely be up for pairing with you. Sadly, Docker setup is something I theoretically know, but always need to puzzle through and refer back to documentation, every time. Which isn't a bad thing, but weird how it doesn't stick in my head like other stuff.